engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -73,6 +79,10 @@ public String apply(@Nullable Level input)
     }
   };
 
+  static {
```

Comment:
Please check log4j java doc: "The MDC is managed on a per *thread* basis. A child thread automatically inherits a copy of the mapped diagnostic context of its parent."